Optimize and align validate_object methods across handlers

obenland opened the pull request:
Also fixes a potential fatal error for `Undo` activities from brid.gy. 

## Proposed changes:
* Optimize `validate_object` methods in Accept, Create, Reject, Undo, and Quote_Request handlers for better performance
* Replace expensive `array_diff()` + `array_keys()` operations with direct `isset()` checks
* Simplify object validation logic from `empty() || !is_array()` to `!isset() || !is_array()`
* Remove unnecessary `is_wp_error()` checks (request parameter is always `WP_REST_Request` in actual usage)
* Rename `$json_params` to `$activity` for improved code readability
* Align validation patterns across all handlers for consistency

### Other information:

- [x] Have you written new tests for your changes, if applicable?

All existing tests pass. Removed one obsolete test that was checking for an impossible scenario (WP_Error as request object).

## Testing instructions:

* Run `npm run env-test` to verify all PHP tests pass
* Verify that inbox validation still works correctly by sending various ActivityPub activities to the inbox

### Changelog entry

<!-- Skip Changelog -->

obenland committed:
Optimize and align validate_object methods across handlers

Performance improvements:
- Replace array_diff() + array_keys() with isset() checks (faster, no temporary arrays)
- Simplify object validation from empty() || !is_array() to !isset() || !is_array()
- Remove unnecessary is_wp_error() checks (request is always WP_REST_Request)

Code quality improvements:
- Rename $json_params to $activity for better readability
- Align validation patterns across all handlers for consistency
- Improve docblock formatting in REST controllers

All existing tests pass.

dependabot[bot] committed:
Bump @wordpress/editor from 14.31.0 to 14.32.0 (#2283)

pfefferle commented on includes/handler/class-undo.php:
```diff
@@ -85,36 +85,25 @@ public static function handle_undo( $activity, $user_id ) {
-			'actor',
-			'object',
-		);
+		if ( ! \is_array( $activity['object'] ) ) {
```
Can we first merge https://github.com/Automattic/wordpress-activitypub/pull/2284 and then have your changes afterwards? My PR adds support for URI-Objects at least for `Like`s and `Announce`s.

pfefferle commented on includes/handler/class-accept.php:
```diff
@@ -72,36 +72,25 @@ public static function handle_accept( $accept, $user_id ) {
+		}
 
-		if ( ! empty( \array_diff( $required_object_attributes, \array_keys( $json_params['object'] ) ) ) ) {
+		if ( ! isset( $activity['object']['id'], $activity['object']['type'], $activity['object']['actor'], $activity['object']['object'] ) ) {
```
Have you changed it to `isset` because of any performance improvements?

I think the current version is way more readable and extensible then a huge list of issets on different levels on array keys!?

pfefferle replied:
ok, seems to be a performance thing!

obenland replied:
Yup!

dependabot[bot] committed:
Bump @wordpress/env from 10.31.0 to 10.32.0 (#2280)

dependabot[bot] committed:
Bump @wordpress/edit-post from 8.31.0 to 8.32.0 (#2282)

pfefferle commented:
> Remove unnecessary is_wp_error() checks (request parameter is always WP_REST_Request in actual usage)

The signature verification returns a `WP_Error` https://github.com/Automattic/wordpress-activitypub/blob/41efe963bbc658f18eae718a46a3eafb5069fd80/includes/rest/class-server.php#L74 does that have no impact on the `validate_object` verification?

pfefferle commented:
Ok, `$request` is always a `WP_REST_Request`. Is the signature verification called afterwards? Is there a way to run the signature verification earlier? We would save some time/resources then, because we wouldn't have to trigger the `validate_object` at all.

obenland committed:
Fix Prettify workflow for pull requests from forks (#2277)

obenland committed:
Tests: Avoid using `runInSeparateProcess` where possible (#2275)

pfefferle committed:
Update builds (again) (#2285)

obenland commented:
We could move it to `rest_pre_dispatch` but at that point the request has not been matched with a handler yet, so would run for every REST API request. Currently every endpoint that needs signature verification specifies it for itself, which we would lose, too.

FWIW, I don't think it's a big deal, eventually a request has to pass all of these tests anyway.

pfefferle commented:
Then let's keep it as is!

obenland committed:
Reorganize test directory structure (#2289)

Jiwoon-Kim committed:
Fix: issue #2286 – typo, improved quoting explanation, add missing i18n in advanced settings (#2290)

pfefferle committed:
Use `bp_members_get_user_url` if available for author URLs (#2292)

pfefferle committed:
Exclude build asset files from PHP GitHub Actions triggers (#2293)

pfefferle commented:
I think it is fine to merge (when the conflicts are resolved), I will update the validation callback through #2295 then.

pfefferle committed:
Merge branch 'trunk' into optimize-validate-object-functions

pfefferle committed:
Remove test for validate_object with WP_Error

Deleted the test_validate_object_with_wp_error method from the quote request test class, possibly because the scenario is no longer relevant or covered elsewhere.
